webpack.config.js

```diff
@@ -0,0 +1,41 @@
+const path = require("path");
+const webpack = require("webpack");
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+
+module.exports = {
+  entry: "./src/client/index.js",
+  mode: "development",
+  module: {
+    rules: [
+      {
+        test: /\.(js|jsx)$/,
+        exclude: /(node_modules)/,
+        loader: "babel-loader",
+        options: { presets: ["@babel/env"] }
+      },
+      {
+        test: /\.css$/,
+        use: ["style-loader", "css-loader"]
+      }
+    ]
+  },
+  resolve: { extensions: ["*", ".js", ".jsx"] },
+  output: {
+    path: path.resolve(__dirname, "dist/"),
+    publicPath: "/dist/",
+    filename: "bundle.js"
+  },
+  devServer: {
+    contentBase: path.join(__dirname, 'src/client/static'),
+    port: process.env.CLIENT_PORT || 3000,
+    publicPath: '/',
+    hotOnly: true
+  },
+  plugins: [
+    new webpack.HotModuleReplacementPlugin(),
```

Comment:
HMR allows most changes in a webpack dev environment to be loaded "on demand" - so when you save a change to a file, if that file was a component in react for instance, it would replace all components that had used the HMR version with the updated code - without "refreshing" the page

Reply:
That's cool!